en/test_and_ci/unit_tests.md

```diff
@@ -0,0 +1,128 @@
+# Unit Tests
+
+PX4 provides a simple base [Unittest-class](https://github.com/PX4/Firmware/blob/master/src/include/unit_test.h). Each developer is encouraged to write unit tests in the process of adding a new feature to the PX4 framework.
+
+## Writing a Test
+
+1. Create a new .cpp file within [tests](https://github.com/PX4/Firmware/tree/master/src/systemcmds/tests) with name `test_[description].cpp`. 
+
+2. Within `test_[description].cpp` include the base unittest-class `<unit_test.h>` and all files required to write a test for the new feature. 
+
+3. Within `test_[description].cpp` create a class `[Description]Test` that inherits from `UnitTest`.
+4. Within `[Description]Test` class declare the public method `virtual bool run_tests()`.
+
+5. Within `[Description]Test` class declare all private methods required to test the feature in question (test1(), test2(),...).
+
+6. Within `test_[description].cpp` implement the `run_tests()` method where each test[1,2,...] will be run.
+
+7. Within `test_[description].cpp`, implement the various tests.
+
+8. At the bottom within `test_[description].cpp` declare the test.
+
+```
```

Comment:
This block should indent 3 spaces so it renders as part of the preceding bullet.